build-system/server/new-server/transforms/sxg/sxg-transform.ts

```diff
@@ -0,0 +1,52 @@
+/**
+ * Copyright 2020 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {PostHTML} from 'posthtml';
+import {URL} from 'url';
+import {isJsonScript, isValidScript} from '../utilities/script';
+import {OptionSet} from '../utilities/option-set';
+
+/**
+ * Returns a function that will transform script node sources into their sxg counterparts.
+ * @param options
+ */
+export default function(options: OptionSet = {}): (tree: PostHTML.Node) => void {
+  return function(tree: PostHTML.Node) {
+    tree.match({tag: 'script'}, (script) => {
+      // Make sure that isJsonScript is used before `isValidScript`. We bail out
+      // early if the ScriptNofe is of type="application/json" since it wouldn't
+      // have any src url to modify.
+      if (isJsonScript(script)) {
+        return script;
+      }
+
+      if (!isValidScript(script)) {
```

Comment:
nit: combine the two above a la `if (isJsonScript(script) || !isValidScript(script)) return script;` Will still short-circuit before `isValidScript`

Reply:
noted. ill make this refactor in https://github.com/ampproject/amphtml/pull/30292 as there 2 other spots i can make this change

Reply:
also the reason why i split them apart is that `isValidScript` does type narrowing when its true so `isJsonScript` is needed to be always first. i was thinking if they are in the same conditional statement someone might think that its possible to reorder them